src/output-json-alert.c

```diff
@@ -715,10 +726,6 @@ static int AlertJson(ThreadVars *tv, JsonAlertLogThread *aft, const Packet *p)
             EvePacket(p, jb, 0);
         }
 
-        if (have_xff_ip && xff_cfg->flags & XFF_EXTRADATA) {
```

Comment:
here it did not depend on `LOG_JSON_APP_LAYER` being set, so this seems like a bigger change than just logging it in a different object

Reply:
Do you mean we want to keep `xff` in root ?

Reply:
the point of my comment is that in this change the xff will not be logged where before it was

Reply:
Indeed.

But beyond that, I feel unsure about where this `xff` should be : stay at the root, or move to http...
What is your feeling on this ?

Reply:
Thinking back to this :
- `xff` is part of `http` for http and fileinfo, not for alert cf `jq 'select(.http.xff) | .event_type' tests/*/*/eve.json`
- If we want `xff` to be part of `http` object in json, I think it is expected that if we do not have `LOG_JSON_APP_LAYER`, we should not have `http` and thus we should not have `http.xff`
- If we want to keep it even without `LOG_JSON_APP_LAYER`, it is fine to leave it at the root

Anyone has an opinion where this `xff` should be ?

Reply:
cc @victorjulien ?

Reply:
Not sure. @jasonish?

Reply:
Tempted to leave it as is.

The xff stuff predates eve where it was just extra data on an alert, so it should probably be kept even in the absence of of metadata logging on alerts.

But I think the ideal solution would be to log a partial http metadata section if we have xff, but metadata logging is off..  So you'd end up with an alert like:
```
{
  "timestamp": "2018-02-16T20:49:12.820418+0000",
  "flow_id": 1646695893454897,
  "pcap_cnt": 9,
  "event_type": "alert",
  "src_ip": "82.165.177.154",
  "src_port": 80,
  "dest_ip": "10.16.1.11",
  "dest_port": 46676,
  "proto": "TCP",
  "tx_id": 0,
  "alert": {
    "action": "allowed",
    "gid": 1,
    "signature_id": 1,
    "rev": 1,
    "signature": "TEST RULE",
    "category": "",
    "severity": 3
  },
  "http": {
    "xff": "10.2.2.2, 10.3.3.3"
  },
  "app_proto": "http",
}
```

If metadata was enabled, you'd then see the full http section of:

```
"http": {
    "hostname": "www.testmyids.com",
    "url": "/",
    "http_user_agent": "curl/7.55.1",
    "xff": "10.2.2.2, 10.3.3.3",
    "http_content_type": "text/html",
    "http_method": "GET",
    "protocol": "HTTP/1.1",
    "status": 200,
    "length": 39
  },
```

(sample from SV http-xff-eve-reverse-extra-data)

Reply:
Done in https://github.com/OISF/suricata/pull/6904